spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/sql/translation/EdcQueryException.java

```diff
@@ -0,0 +1,23 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.sql.translation;
+
+import org.eclipse.dataspaceconnector.spi.EdcException;
+
+public class EdcQueryException extends EdcException {
```

Comment:
I don't know if I would add another exception, anyway this would need to be handled in the `EdcApiExceptionMapper` to return the expected 400.
Could be that the `query` method of every service returns a `ServiceResult<Collection<T>>`, in this way the error could be marked as `ServiceFailure.Reason.BAD_REQUEST` and be handled correctly to return a 400 by the controller

Reply:
returning a `ServiceResult` is probably better.

Reply:
created https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/1654